runtime/lua/vim/_defaults.lua

```diff
@@ -107,22 +107,51 @@ do
           vim.inspect(cmd.cmd)
         )
       end
+      return err
+    end
 
-      if err then
-        vim.notify(err, vim.log.levels.ERROR)
+    --- Gets the URL at cursor, if any.
+    local function get_url()
+      local current_node =
+        vim.treesitter.get_node { ignore_injections = false, lang = 'markdown_inline' }
+
+      while current_node do
```

Comment:
should it fallback to `vim.fn.expand('<cfile>')` ? or is that implied already?

Reply:
It is not implied, but I'm not sure it should fallback. When pressing `gx` on an empty line on master it opens the file explorer of the directory where the file is. This is because of `vim.fn.expand('<cfile>')` resolves to empty string in that case. If this is the intended behavior I can put back `vim.fn.expand('<cfile>')`.

Reply:
my question is more about what happens if `vim.treesitter.get_node { ignore_injections = false, lang = 'markdown_inline' }` returns nothing? `get_url()` should support all kinds of buffers, not only markdown. so presumably `<cfile>` is a way to do that

Reply:
I'll add `<cfile>` for now and evaluate the behavior after.

Reply:
Yes, you're right. I tested it with this pr and it fails for "normal" urls. Added `<cfile>` as fallback.